### PR TITLE
[XLA] Extra VLOG for PTXAS calls 

### PR DESCRIPTION
**Pull request by nouiz**
This PR does a few changes related to PTXAS:
- print the PTXAS command line used via VLOG
- It will also output the verbose output of PTXAS as already asked on line 210, but not displayed.
- Add the TF_EXTRA_PTXAS_OPTIONS environment variable that allows to pass extra parameter to PTXAS.

I tag this XLA as to my knowledge this code is only used by XLA.

### CODE REVIEW

File: tensorflow/stream_executor/gpu/asm_compiler.cc [hunk 1]
```diff
@@ -237,6 +249,16 @@ port::StatusOr<std::vector<uint8>> CompileGpuAsm(int cc_major, int cc_minor,
   return cubin_vector;
 }
 
+char* GetTFExtraPTXOptions() {
+  static bool fetched = false;
+  static char* env = nullptr;
+  if (!fetched) {
+    env = getenv("TF_EXTRA_PTXAS_OPTIONS");
```
Comment (timshen91) [inline tensorflow/stream_executor/gpu/asm_compiler.cc:260]: If it's only for XLA, the flag should go through XLA_FLAGS. See https://github.com/tensorflow/tensorflow/blob/13645c7fd2a7c151b82a9bfa144b0e682122ed7b/tensorflow/compiler/xla/xla.proto#L25

`GpuAsmOpts` already has a member `extra_flags`, which can be plumbed through XLA_FLAGS.

**Review by timshen91 — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@nouiz  Can you please check @timshen91's comments and keep us posted. Thanks!

**Comment by nouiz** [comment]
Will do. But it could take a few days.

**Comment by nouiz** [comment]
Updated per instruction. I tested manually with run_hlo_modules and it behave as expected.

Note, I tested with replay_computation and now the XLA_FLAGS aren't passed correctly for it. So it doesn't behave correctly. But this isn't related to this PR.

**Comment by nouiz** [comment]
run_hlo_module seems to replace replay_computation. As replay_computation is partially broken, why not just remove it and point to the new tools?

File: tensorflow/compiler/xla/service/gpu/stream_executor_util.cc [hunk 2]
```diff
@@ -222,9 +222,13 @@ Status ExecuteKernelOnStream(const se::KernelBase& kernel,
 }
 
 se::GpuAsmOpts PtxOptsFromConfig(const HloModuleConfig& hlo_module_config) {
+  string extra_string = hlo_module_config.debug_options().xla_gpu_gpuasm_extra_flags();
+  std::vector<std::string> extra_flags;
+  extra_flags = absl::StrSplit(extra_string, " ", absl::SkipEmpty());
```
Comment (timshen91) [inline tensorflow/compiler/xla/service/gpu/stream_executor_util.cc:227]: Can you use comma delimiter? Using spaces makes it hard in mix with other options, e.g. XLA_FLAGS="--xla_gpu_gpuasm_extra_flags=a b c --xla_hlo_profile"

File: tensorflow/compiler/xla/xla.proto [hunk 3]
```diff
@@ -287,7 +287,10 @@ message DebugOptions {
   // memory, or have bugs.
   bool xla_gpu_unsafe_fallback_to_driver_on_ptxas_not_found = 138;
 
-  // Next id: 141
+  // Extra parameters to pass the GPU assembler.
+  string xla_gpu_gpuasm_extra_flags = 141;
```
Comment (timshen91) [inline tensorflow/compiler/xla/xla.proto:291]: xla_gpu_asm_extra_flags

**Review by timshen91 — COMMENTED** [review]
> Updated per instruction. I tested manually with run_hlo_modules and it behave as expected.
> 
> Note, I tested with replay_computation and now the XLA_FLAGS aren't passed correctly for it. So it doesn't behave correctly. But this isn't related to this PR.

I just tried and at least XLA_FLAGS="--xla_dump_to=..." works.

**Review by timshen91 — APPROVED** [review]

**Pull request merged**